### PR TITLE
Use dashboard store in view mode also

### PR DESCRIPTION
**Pull request by LukeTillman**
This just removes the `useDashboard` hook and some of the weird state flow where I screwed up and told Eunice we should be passing the dashboard spec around everywhere. 😂  Instead, it just uses the `layouts` and `panels` state already in the store for both view and edit modes. I haven't tried to do anything "smart" yet (like more granular selectors to help with re-rendering), just simplified the mess I made so we can start doing that.

### CODE REVIEW

File: ui/dashboards/src/context/DashboardProvider/DashboardProvider.tsx [hunk 1]
```diff
@@ -42,18 +42,6 @@ export function useEditMode() {
   return useDashboardStore(({ isEditMode, setEditMode }) => ({ isEditMode, setEditMode }));
 }
 
-export function useDashboard() {
```
Comment (eunicorn) [inline ui/dashboards/src/context/DashboardProvider/DashboardProvider.tsx:45]: should we keep this for the export feature?

File: ui/dashboards/src/components/GridLayout/GridLayout.tsx [hunk 2]
```diff
@@ -13,17 +13,18 @@
...
 import { useEditMode } from '../../context';
 import { GridTitle } from './GridTitle';
+import { GridItemContent } from './GridItemContent';
 
 const ResponsiveGridLayout = WidthProvider(Responsive);
 
 export interface GridLayoutProps extends BoxProps {
   groupIndex: number;
   definition: GridDefinition;
-  renderGridItemContent: (definition: GridItemDefinition, itemIndex: number) => React.ReactNode;
```
Comment (eunicorn) [inline ui/dashboards/src/components/GridLayout/GridLayout.tsx:26]: i've been wanting to remove this for so long 😆

Comment (LukeTillman) [reply ui/dashboards/src/components/GridLayout/GridLayout.tsx:26]: Hahahaha, that was back when I thought `GridLayout` might be more generic/reusable on its own, but good to just "embrace the store".

File: ui/dashboards/src/context/DashboardProvider/DashboardProvider.tsx (hunk 1 above)
Comment (LukeTillman) [reply ui/dashboards/src/context/DashboardProvider/DashboardProvider.tsx:45]: I figure we should probably just build a "getDashboardJson" or something into the store for export (whenever we get around to it).

**Pull request merged**